plugins/catalog-backend-module-ldap/src/ldap/util.ts

```diff
@@ -24,3 +25,24 @@ import { Error as LDAPError } from 'ldapjs';
 export function errorString(error: LDAPError) {
   return `${error.code} ${error.name}: ${error.message}`;
 }
+
+/**
+ * Maps a single-valued attribute to a consumer
+ * @param entry The LDAP source entry
+ * @param vendor The LDAP vendor
+ * @param attributeName The source attribute to map
```

Comment:
Mention that if it's undefined, nothing is done.

It's a bit unfortunate to have this kind of a concern-leaking thing among the exports ... it was only made this way to make it a oneliner when dealing with unset config values. Ah well, maybe others will also want to mirror that behaviour if they want to replace the builtins while also obeying config.

Reply:
Let's leave it in for now. The alternative would be to instead export a separate function (e.g. `mapStringAttribute`) which has a non-optional parameter and is called by the internal `mapStringAttr` perhaps. But don't worry about it, it's not really a big deal.